### ISSUE TITLE
using a when statement on an include only applies to first task in include file

### ISSUE THREAD

**Issue by zer0glitch** [ansible/ansible#18448]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

- include

##### ANSIBLE VERSION

```
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

host_key_checking=false

##### OS / ENVIRONMENT

CentOS Linux release 7.2.1511 (Core) 

##### SUMMARY

When there is a top level playbook that will include a sub level playbook based upon a variable in the inventory file, the when statement only applies to the first task in the sub playbook..  this only applies to 2.2.0.0.  This worked without issue in 2.1.2.0

##### STEPS TO REPRODUCE

inventory
```
myserver  ansible_connection=local

[mygroup]
myserver

[mygroup:vars]
include_if_true=false
```

top.yml
```

---

- include: includes/include_if_true.yml
  when: include_if_true

- include: includes/include_if_not_true.yml
  when: not include_if_true

- hosts: all
  tasks:
  - debug: msg="Just printing to expand the example"
```
includes/always_include.yml
```

---

- hosts: all
  tasks:

  - name: print message always
    debug: msg="Value of includ if true {{ include_if_true }}"

  - name: print message always 2
    debug: msg="Value of includ if true {{ include_if_true }}"

```

includes/include_if_not_true.yml
```

---

- include: always_include.yml

- hosts: all
  tasks:

  - name: print message not true
    debug: msg="Included if not true"
    when: inventory_hostname in groups['mygroup']

  - name: print message not tru2e
    debug: msg="Included if not true2"
    when: inventory_hostname in groups['mygroup']

```

includes/include_if_true.yml
```

---

- hosts: all
  tasks:

  - name: print message true
    debug: msg="Included if true"
    when: inventory_hostname in groups['mygroup']

  - name: print message true2
    debug: msg="Included if true2"
    when: inventory_hostname in groups['mygroup']

```

```
ansible-playbook  top.yml -i inventory -vv

```

##### EXPECTED RESULTS

```
Using /etc/ansible/ansible.cfg as config file

PLAYBOOK: top.yml **************************************************************
4 plays in top.yml

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [myserver]

TASK [print message true] ******************************************************
task path: /vagrant/playbooks/include_bug/includes/include_if_true.yml:6
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [print message true2] *****************************************************
task path: /vagrant/playbooks/include_bug/includes/include_if_true.yml:10
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [myserver]

TASK [print message always] ****************************************************
task path: /vagrant/playbooks/include_bug/includes/always_include.yml:6
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [print message always 2] **************************************************
task path: /vagrant/playbooks/include_bug/includes/always_include.yml:9
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [myserver]

TASK [print message not true] **************************************************
task path: /vagrant/playbooks/include_bug/includes/include_if_not_true.yml:8
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [print message not tru2e] *************************************************
task path: /vagrant/playbooks/include_bug/includes/include_if_not_true.yml:12
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [myserver]

TASK [debug] *******************************************************************
task path: /vagrant/playbooks/include_bug/top.yml:11
ok: [myserver] => {
    "msg": "Just printing to expand the example"
}

PLAY RECAP *********************************************************************
myserver                   : ok=5    changed=0    unreachable=0    failed=0   

```

##### ACTUAL RESULTS

```
sing /etc/ansible/ansible.cfg as config file

PLAYBOOK: top.yml **************************************************************
4 plays in top.yml

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [myserver]

TASK [print message true] ******************************************************
task path: /vagrant/playbooks/include_bug/includes/include_if_true.yml:6
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [print message true2] *****************************************************
task path: /vagrant/playbooks/include_bug/includes/include_if_true.yml:10
ok: [myserver] => {
    "msg": "Included if true2"
}

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [myserver]

TASK [print message always] ****************************************************
task path: /vagrant/playbooks/include_bug/includes/always_include.yml:6
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [print message always 2] **************************************************
task path: /vagrant/playbooks/include_bug/includes/always_include.yml:9
ok: [myserver] => {
    "msg": "Value of includ if true false"
}

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [myserver]

TASK [print message not true] **************************************************
task path: /vagrant/playbooks/include_bug/includes/include_if_not_true.yml:8
skipping: [myserver] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

TASK [print message not tru2e] *************************************************
task path: /vagrant/playbooks/include_bug/includes/include_if_not_true.yml:12
ok: [myserver] => {
    "msg": "Included if not true2"
}

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [myserver]

TASK [debug] *******************************************************************
task path: /vagrant/playbooks/include_bug/top.yml:11
ok: [myserver] => {
    "msg": "Just printing to expand the example"
}

PLAY RECAP *********************************************************************
myserver                   : ok=8    changed=0    unreachable=0    failed=0   
```

```
see above
```

**Comment by jctanner** [comment]
@jimi-c is this related to https://github.com/ansible/ansible/issues/18318 ?

**Comment by jimi-c** [comment]
I would say high probability it is related, I'll test now.

**Comment by jimi-c** [comment]
This appears to be resolved on the latest devel and stable-2.2 branches, so I'll close it now.  This will also be included in the next `2.2.1.0` release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
